include/rdma/fi_domain.h

```diff
@@ -152,6 +153,7 @@ struct fi_mr_attr {
 		int		neuron;
 		int		synapseai;
 	} device;
+	void			*hmem_data;
```

Comment:
Hmmmm I considered that before opening this PR. Reusing `context` is problematic based on the current man page:
> **context**
Application context associated with asynchronous memory registration operations. This value is returned as part of any asynchronous event associated with the registration. This field is ignored for synchronous registration calls.

Conceptually, storing the gdrcopy handle in `context` has the same problem as in `device` before - neither `context` or `device` field is intended for the handle.

Since we are not yet ready to make `hmem_data` externally useful, I propose we make a similar disclaimer to `offset`, i.e.
> The offset field is reserved for future use and must be 0.

->

> The hmem_data field is reserved for future use and must be null.

Do this make sense?

Reply:
This works.  I can see it being useful once properly defined.

Reply:
Thank you! I will address the comments and update.